src/Interfaces/GatewayInterface.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace CraigPaul\Moneris\Interfaces;
+
+interface GatewayInterface
+{
+    /**
+     * Capture a pre-authorized a transaction.
+     * @param \CraigPaul\Moneris\Transaction|string $transaction
+     * @param string|null $order
+     * @param mixed|null $amount
+     * @return \CraigPaul\Moneris\Response
+     */
+    public function capture ($transaction, $order = null, $amount = null);
+
+    /**
+     * Create a new Vault instance.
+     * @return \CraigPaul\Moneris\Vault
+     */
+    public function cards ();
+
+    /**
+     * Pre-authorize a purchase.
+     * @param array $params
+     * @return \CraigPaul\Moneris\Response
+     */
+    public function preauth (array $params = []);
```

Comment:
Why not type hint like
`public function preauth (array $params = []): Response;`

Reply:
I'm not adding any type hinting for now, just trying to ship something that works. I'm planning to change these over time but honestly the value proposition is kinda low for typehints at the moment.